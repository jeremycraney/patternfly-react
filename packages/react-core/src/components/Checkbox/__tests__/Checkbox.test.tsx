import * as React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { Checkbox } from '../Checkbox';

const props = {
  onChange: jest.fn(),
  isChecked: false
};

test('controlled', () => {
  const view = render(<Checkbox isChecked id="check" aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('controlled - 3rd state', () => {
  const view = render(<Checkbox isChecked={null} id="check" aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('uncontrolled', () => {
  const view = render(<Checkbox id="check" aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('isDisabled', () => {
  const view = render(<Checkbox id="check" isDisabled aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('label is string', () => {
  const view = render(<Checkbox label="Label" id="check" isChecked aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('label is function', () => {
  const functionLabel = () => <h1>Header</h1>;
  const view = render(<Checkbox label={functionLabel()} id="check" isChecked aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('label is node', () => {
  const view = render(<Checkbox label={<h1>Header</h1>} id="check" isChecked aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('passing class', () => {
  const view = render(<Checkbox label="label" className="class-123" id="check" isChecked aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('passing HTML attribute', () => {
  const view = render(<Checkbox label="label" aria-labelledby="labelId" id="check" isChecked aria-label="check" />);
  expect(view.container).toMatchSnapshot();
});

test('passing description', () => {
  const view = shallow(<Checkbox id="check" label="checkbox" description="Text description..." />);
  const descriptionEl = view.find('span[className="pf-c-check__description"]');
  expect(descriptionEl.length).toBe(1);
  expect(descriptionEl.text()).toBe('Text description...');
});

test('passing body', () => {
  const view = shallow(<Checkbox id="check" label="checkbox" body="This is where custom content goes." />);
  const bodyEl = view.find('span[className="pf-c-check__body"]');
  expect(bodyEl.length).toBe(1);
  expect(bodyEl.text()).toBe('This is where custom content goes.');
});

test('checkbox passes value and event to onChange handler', () => {
  const newValue = true;
  const event = {
    currentTarget: { checked: newValue }
  };
  const view = shallow(<Checkbox id="check" {...props} aria-label="check" />);
  view.find('input').simulate('change', event);
  expect(props.onChange).toBeCalledWith(newValue, event);
});

test('should throw console error when no id is given', () => {
  const myMock = jest.fn();
  global.console = { ...global.console, error: myMock };
  render(<Checkbox {...props} />);
  expect(myMock).toBeCalled();
});
