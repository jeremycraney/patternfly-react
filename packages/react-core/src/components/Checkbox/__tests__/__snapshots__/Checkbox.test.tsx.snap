// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`controlled - 3rd state 1`] = `
<div>
  <div
    class="pf-c-check pf-m-standalone"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      class="pf-c-check__input"
      data-ouia-component-id="OUIA-Generated-Checkbox-2"
      data-ouia-component-type="PF4/Checkbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`controlled 1`] = `
<div>
  <div
    class="pf-c-check pf-m-standalone"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      checked=""
      class="pf-c-check__input"
      data-ouia-component-id="OUIA-Generated-Checkbox-1"
      data-ouia-component-type="PF4/Checkbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`isDisabled 1`] = `
<div>
  <div
    class="pf-c-check pf-m-standalone"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      class="pf-c-check__input"
      data-ouia-component-id="OUIA-Generated-Checkbox-4"
      data-ouia-component-type="PF4/Checkbox"
      data-ouia-safe="true"
      disabled=""
      id="check"
      type="checkbox"
    />
  </div>
</div>
`;

exports[`label is function 1`] = `
<div>
  <div
    class="pf-c-check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      checked=""
      class="pf-c-check__input"
      data-ouia-component-id="OUIA-Generated-Checkbox-6"
      data-ouia-component-type="PF4/Checkbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
    <label
      class="pf-c-check__label"
      for="check"
    >
      <h1>
        Header
      </h1>
    </label>
  </div>
</div>
`;

exports[`label is node 1`] = `
<div>
  <div
    class="pf-c-check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      checked=""
      class="pf-c-check__input"
      data-ouia-component-id="OUIA-Generated-Checkbox-7"
      data-ouia-component-type="PF4/Checkbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
    <label
      class="pf-c-check__label"
      for="check"
    >
      <h1>
        Header
      </h1>
    </label>
  </div>
</div>
`;

exports[`label is string 1`] = `
<div>
  <div
    class="pf-c-check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      checked=""
      class="pf-c-check__input"
      data-ouia-component-id="OUIA-Generated-Checkbox-5"
      data-ouia-component-type="PF4/Checkbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
    <label
      class="pf-c-check__label"
      for="check"
    >
      Label
    </label>
  </div>
</div>
`;

exports[`passing HTML attribute 1`] = `
<div>
  <div
    class="pf-c-check"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      aria-labelledby="labelId"
      checked=""
      class="pf-c-check__input"
      data-ouia-component-id="OUIA-Generated-Checkbox-9"
      data-ouia-component-type="PF4/Checkbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
    <label
      class="pf-c-check__label"
      for="check"
    >
      label
    </label>
  </div>
</div>
`;

exports[`passing class 1`] = `
<div>
  <div
    class="pf-c-check class-123"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      checked=""
      class="pf-c-check__input"
      data-ouia-component-id="OUIA-Generated-Checkbox-8"
      data-ouia-component-type="PF4/Checkbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
    <label
      class="pf-c-check__label"
      for="check"
    >
      label
    </label>
  </div>
</div>
`;

exports[`uncontrolled 1`] = `
<div>
  <div
    class="pf-c-check pf-m-standalone"
  >
    <input
      aria-invalid="false"
      aria-label="check"
      class="pf-c-check__input"
      data-ouia-component-id="OUIA-Generated-Checkbox-3"
      data-ouia-component-type="PF4/Checkbox"
      data-ouia-safe="true"
      id="check"
      type="checkbox"
    />
  </div>
</div>
`;
